app/models/bench.rb

```diff
@@ -23,6 +24,22 @@ class Bench < ApplicationRecord
            class_name: 'RequestDistribution',
            inverse_of: :bench,
            dependent: :restrict_with_error
+  # Checks
+  def dimensions_must_be_positive
+    return unless dimensions
+
+    return unless dimensions.any? { |d| d <= 0 }
+
+    errors.add(:dimensions, 'each dimension must be greater than 0')
+  end
+
+  def positions_must_be_positive
+    return unless positions
+
+    return unless positions.any?(&:negative?)
+
+    errors.add(:positions, 'each position must be positive')
+  end
```

Comment:
très bien ces validations custom ! :)

bon, c'est du pinaillage, mais je trouve tout de même un peu déroutant qu'elles aient le même nom mais ne vérifient pas tout à fait la même chose ; peut être devrait-on préciser dans le nom de la première que les dimensions doivent être strictement positives ? Pour rendre plus explicite le traitement du zéro.

Quelque chose comme ça, par exemple :
```suggestion
  def dimensions_must_be_strictly_positive
    return unless dimensions
    return if dimensions.all?(&:positive)

    errors.add(:dimensions, 'each dimension must be greater than 0')
  end

  def positions_must_be_positive
    return unless positions
    return unless positions.any?(&:negative?)

    errors.add(:positions, 'each position must be greater or equal to 0')
  end
```
(si tu veux faire ça, n'oublie pas de modifier le nom de la méthode également dans le `validate` un peu plus haut !)